tldw_Server_API/app/core/Evaluations/recipes/rag_answer_quality_execution.py

```
@@ -0,0 +1,1155 @@
+    raw_contexts = None
+    for field_name in ("retrieved_contexts", "inline_contexts", "contexts", "source_contexts"):
+        value = sample.get(field_name)
+        if value:
+            raw_contexts = value
+            break
+    if raw_contexts is None and isinstance(sample.get("context"), Mapping):
+        raw_contexts = [sample.get("context")]
+    if raw_contexts is None:
+        return []
+    contexts: list[dict[str, Any]] = []
+    for item in list(raw_contexts):
```

Ruff (0.15.7)

[warning] Do not cast an iterable to `list` before iterating over it

Remove `list()` cast

(PERF101)